forc-plugins/forc-crypto/src/content.rs

```diff
@@ -0,0 +1,63 @@
+use std::str::FromStr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::{
+    fs::read,
+    io::{self, Read},
+};
+
+static STDIN_HAS_BEEN_USED: AtomicBool = AtomicBool::new(false);
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Content {
+    Path(std::path::PathBuf, Vec<u8>),
+    Raw(Vec<u8>),
+}
+
+impl Content {
+    pub fn from_hex_or_raw(input: Vec<u8>) -> Vec<u8> {
```

Comment:
I have little concerns about this. 

1. The name makes the impression that this will be returning a `Content`, but it returns an `Vec<u8>`. It is a little confusing.

2. TL;DR: How do we differentiate the prefix bytes from actual bytes if they are `[48, 120]`? IIUC, this function tries to see if the given input is `Raw` or in `Hex` form by checking whether text converted version of the input starts with `0x`. I am not sure how we collect this `Raw` input from the user and why it differs from the hex. My intuition is that they are in the following forms:

*For `0x1a2d`*:

- Hex input: `[48, 120, 49, 97, 50, 100]`
- Raw input: `[26, 45]`

and the expected outcome for this function is to return the same thing for both inputs provided above which is: `[26, 45]`. Is this correct?

If this is correct what happens if the raw input I want to provide starts with byte equivalents of `0x`. For example what happens if the raw input is `[48, 120, 100, 100]`. This will be translated to `0xd` and first if branch will be taken. The output of the function would be `[221]` because 0x will be trimmed and `dd` will be decoded by `hex` crate to `[221]` but the expected output was `[48, 120, 100, 100]`?

Reply:
@kayagokalp I improved the logic. Although it is taking a vector of u8, it is expecting it to be a valid UTF-8 string. Anything else will error. Binary data should be encoded as hexadecimal strings.